### PR TITLE
feat(tools): add return_direct parameter to skip LLM processing of tool output

### LINKED ISSUE #1240: Implement "Return Direct" Functionality for Tool Outputs
**Problem:** When using certain tools that generate detailed or specific text output (e.g., logs, structured data, code snippets), the agent currently attempts to process and rewrite this output. This process is detrimental as it frequently strips away critical information, context, or formatting that is essential for the user to interpret the tool's results correctly. The agent's modified response becomes useless.

**Desired Feature:** We request a mechanism (e.g., a parameter when calling or defining a tool, or a specific agent instruction) that allows us to designate that the raw text output from a particular tool execution should be returned *verbatim* as the final response to the user, bypassing the agent's standard response generation/modification process.

**Analogy:** This functionality is analogous to the `return_direct` feature found in other agent frameworks like LangChain, where a specific step's output is designated as the final, unmodified answer.

**Benefit:** This would enable the effective use of tools that produce output which must be presented precisely as generated, preserving all details and preventing data loss caused by agent summarization or rewriting.

### PR DESCRIPTION
**Pull request by giulio-leone**
## Summary
Implements the "Return Direct" feature requested in #1240.

When a Tool has `return_direct=True`, the agent returns the tool output immediately without passing it back through the LLM for processing. This can save up to 50% response time for tools whose output is already the final answer.

## Changes
- Added `return_direct: bool = False` to Tool class
- Modified `process_tool_calls` in ToolCallingAgent to check `return_direct` after tool execution
- Modified `evaluate_python_code` in executor to wrap `return_direct` tools with `FinalAnswerException` (same pattern as `final_answer`)
- Works with both `ToolCallingAgent` and `CodeAgent`

## Usage
```python
class MyTool(Tool):
    name = "my_tool"
    description = "Returns a direct answer"
    inputs = {"query": {"type": "string", "description": "The query"}}
    output_type = "string"
    return_direct = True

    def forward(self, query: str) -> str:
        return f"Answer: {query}"
```

## Tests
Added comprehensive test suite (`tests/test_return_direct.py`) with 11 tests covering:
- Default `return_direct=False` behavior
- `return_direct=True` skips LLM for both ToolCallingAgent and CodeAgent
- Normal tools still go through the LLM
- Edge cases (multiple tool calls, getattr fallback)

Fixes #1240

### CODE REVIEW

File: src/smolagents/tools.py [hunk 1]
```diff
@@ -133,6 +136,7 @@ class Tool(BaseTool):
     inputs: dict[str, dict[str, str | type | bool]]
     output_type: str
     output_schema: dict[str, Any] | None = None
+    return_direct: bool = False
```
[the remark this answers is not in the document]
Comment (giulio-leone) [reply src/smolagents/tools.py]: Fixed: added bool validation for `return_direct` in `validate_arguments()`, consistent with the existing validation pattern for `output_schema`.

[the remark this answers is not in the document]
Comment (giulio-leone) [reply src/smolagents/tools.py]: Fixed: `return_direct` is now included in `to_dict()` (when True) and restored in `from_dict()`, following the same pattern as `output_schema`.

File: src/smolagents/local_python_executor.py [hunk 2]
```diff
@@ -1635,6 +1635,19 @@ def final_answer(*args, **kwargs):  # Allow arbitrary arguments to be passed
...
+        if tool_name != "final_answer" and getattr(tool_func, "return_direct", False):
+
+            def _make_return_direct_wrapper(original_tool):
+                def wrapper(*args, **kwargs):
+                    result = original_tool(*args, **kwargs)
+                    raise FinalAnswerException(result)
+
+                return wrapper
+
+            static_tools[tool_name] = _make_return_direct_wrapper(tool_func)
```
[the remark this answers is not in the document]
Comment (giulio-leone) [reply src/smolagents/local_python_executor.py]: Acknowledged: `return_direct` with remote executors is a known limitation. The current implementation wraps tools in the local executor via `FinalAnswerException`, which only works locally. This is documented as local-only behavior for now; extending to remote executors would require protocol-level changes and is out of scope for this PR.

File: src/smolagents/agents.py [hunk 3]
```diff
@@ -1404,6 +1404,11 @@ def process_single_tool_call(tool_call: ToolCall) -> ToolOutput:
                 level=LogLevel.INFO,
             )
             is_final_answer = tool_name == "final_answer"
+            if not is_final_answer:
+                available_tools = {**self.tools, **self.managed_agents}
+                tool = available_tools.get(tool_name)
+                if tool is not None and getattr(tool, "return_direct", False):
+                    is_final_answer = True
```
[the remark this answers is not in the document]
Comment (giulio-leone) [reply src/smolagents/agents.py]: Fixed: updated the error message to mention `return_direct` tools alongside the final answer tool, so users get clearer feedback when mixing return_direct calls with other tool calls.

File: tests/test_return_direct.py [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+        result = agent.run("test query")
+        assert result == "Direct result for: test"
+
+
+class TestReturnDirectEdgeCases:
+    """Test edge cases for the return_direct feature."""
+
+    def test_return_direct_with_getattr(self):
+        """getattr fallback works for tools without return_direct attribute."""
+        tool = NormalTool()
```
[the remark this answers is not in the document]
Comment (giulio-leone) [reply tests/test_return_direct.py]: Fixed: the test now uses a bare `BareObject` class without the `return_direct` attribute to properly test the `getattr` fallback. Also kept the assertion on `NormalTool` to verify the default value.

File: tests/test_return_direct.py [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+        retries.  We just verify it doesn't crash."""
+        model = FakeToolCallModelMultipleTools()
+        agent = ToolCallingAgent(
+            tools=[ReturnDirectTool(), NormalTool()],
+            model=model,
+            max_steps=2,
+        )
+        # The agent should not crash; it handles the error internally
+        # and continues (or hits max_steps).
+        agent.run("test query")
```
[the remark this answers is not in the document]
Comment (giulio-leone) [reply tests/test_return_direct.py]: Fixed: the test now captures the return value from `agent.run()`. The primary assertion is that execution completes without raising — the agent handles the error internally and either returns a result or hits max_steps.
